alica_engine/include/engine/model/AbstractPlan.h

```diff
@@ -25,8 +25,12 @@ class AlicaEngine;
 class AbstractPlan : public AlicaElement
 {
 public:
-    AbstractPlan(AlicaEngine* ae);
-    AbstractPlan(AlicaEngine* ae, int64_t id);
+    //[[deprecated("It will be removed in the last PR")]]
+    AbstractPlan(AlicaEngine* ae); // TOBE removed
+    //[[deprecated("It will be removed in the last PR")]]
+    AbstractPlan(AlicaEngine* ae, int64_t id); // TOBE removed
+    AbstractPlan(const YAML::Node& config, SubscribeFunction subscribeFunc);
+    AbstractPlan(const YAML::Node& config, SubscribeFunction subscribeFunc, int64_t id);
```

Comment:
authorityTimeInterval could be moved to Plan class. Its only valid for plans and doesn't belong in the base class.
With that none of Behaviour, AbstractPlan, PlanType need to pass around config and subscriber function,

Reply:
that's alright.  
if (_rp->isBehaviour()) {
        return;
}

Reply:
Ok! I add the check on the fly also on
CycleManager::handleAuthorityInfo